### PR TITLE
feat: create key in encryption service for merchant and user

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] New feature

## Description

This PR adds support to

- Create Key for every User and Merchant during User and Merchant Account Create.
- Add an API to transfer Keys to KeyManager
- Add MTLS for KeyManager Client

### Additional Changes
- [x] This PR modifies application configuration/environment variables

## Motivation and Context

 - Create Key in Key Manager for every merchant

### Tests for  Create a Merchant
-
```bash
curl --location 'http://localhost:8080/accounts' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: test_admin' \
--data-raw '{
  "merchant_id": "merchant_1720435957",
  "locker_id": "m0010",
  "merchant_name": "NewAge Retailer",
  "merchant_details": {
    "primary_contact_person": "John Test",
    "primary_email": "JohnTest@test.com",
    "primary_phone": "sunt laborum",
    "secondary_contact_person": "John Test2",
    "secondary_email": "JohnTest2@test.com",
    "secondary_phone": "cillum do dolor id",
    "website": "www.example.com",
    "about_business": "Online Retail with a wide selection of organic products for North America",
    "address": {
      "line1": "1467",
      "line2": "Harrison Street",
      "line3": "Harrison Street",
      "city": "San Fransico",
      "state": "California",
      "zip": "94122",
      "country": "US"
    }
  },
  "return_url": "https://google.com/success",
  "webhook_details": {
    "webhook_version": "1.0.1",
    "webhook_username": "ekart_retail",
    "webhook_password": "password_ekart@123",
    "payment_created_enabled": true,
    "payment_succeeded_enabled": true,
    "payment_failed_enabled": true
  },
  "sub_merchants_enabled": false,
  "metadata": {
    "city": "NY",
    "unit": "245"
  },
  "primary_business_details": [
    {
      "country": "US",
      "business": "default"
    }
  ]
}'
```
 - Verify `/key/create` to Keymanager was succeeded.
 The response status of `keymanager_response` should be 200

Screenshot 2024-07-11 at 5 01 27 PM

### Create Key for User

- Signin with the account and password
```
curl --location 'http://localhost:8080/user/v2/signin?token_only=true' \
--header 'Content-Type: application/json' \
--data-raw '{
 "email": "gagagagag@gagagagagag.com",
    "password": "gagagagag@1234"
}'

```
- Begin TOTP flow
```
curl --location 'http://localhost:8080/user/2fa/totp/begin' \
--header 'Authorization: Bearer spt'
```
- Verify that call to `/key/create` is successful in the logs

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code

### CODE REVIEW

**Review by ArjunKarthik — DISMISSED** [review]

**Review by ArjunKarthik — APPROVED** [review]
LGTM

**Review by ArjunKarthik — DISMISSED** [review]

**Review by jarnura — APPROVED** [review]

**Review by ArjunKarthik — APPROVED** [review]

File: crates/router/src/routes/admin.rs [hunk 1]
```diff
@@ -651,6 +651,27 @@ pub async fn merchant_account_kv_status(
     .await
 }
 
+/// Merchant Account - KV Status
+///
+/// Toggle KV mode for the Merchant Account
```
Comment (racnan) [inline crates/router/src/routes/admin.rs:656]: ??

File: crates/router/src/routes/admin.rs [hunk 2]
```diff
@@ -651,6 +651,27 @@ pub async fn merchant_account_kv_status(
...
+#[instrument(skip_all)]
+pub async fn merchant_account_transfer_keys(
+    state: web::Data<AppState>,
+    req: HttpRequest,
+) -> HttpResponse {
+    let flow = Flow::ConfigKeyFetch;
```
Comment (racnan) [inline crates/router/src/routes/admin.rs:662]: Needs to be changed.

File: crates/common_utils/src/keymanager.rs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+) -> errors::CustomResult<reqwest::Response, errors::KeyManagerClientError>
+where
+    T: serde::Serialize,
+{
+    let client = get_api_encryption_client(state)?;
+    let url = reqwest::Url::parse(&url)
+        .change_context(errors::KeyManagerClientError::UrlEncodingFailed)?;
+
+    client
+        .post(url)
```
Comment (racnan) [inline crates/common_utils/src/keymanager.rs:76]: shouldnt generic function allow `method` to be passed ?

**Review by racnan — APPROVED** [review]
Nit picks

**Review by ArjunKarthik — DISMISSED** [review]

**Review by racnan — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

**Review by racnan — APPROVED** [review]

**Review by lsampras — APPROVED** [review]

**Review by jarnura — APPROVED** [review]

**Pull request merged**